### ISSUE TITLE
Tags: setting `tags_extra_files` not working as expected

### ISSUE THREAD

**Issue by hellt** [squidfunk/mkdocs-material#5914]
### Bug description

When upgraded to 9.2.5-insiders-4.40.1 I started getting the `Plugin 'material/tags' option 'tags_extra_files': Expected type: <class 'str'> but received: <class 'list'>` error, where it was working fine for `insiders 4.36.0`.

Judging by the documentation, the tag_extra_files struct hasn't changed, so I presume there might be an issue.

Here is my tags section:

```yaml
  - tags:
      tags_file: blog/tags.md
      # media.md is a file that only shows posts with `media` tag
      tags_extra_files:
        blog/media.md:
          - media
```

Changing the type to string for tag slugs like this:

```yaml
  - tags:
      tags_file: blog/tags.md
      # media.md is a file that only shows posts with `media` tag
      tags_extra_files:
        blog/media.md: media
```

Gets me past the type check, but, as expected, fails when the tag mapping function kicks in:

```
ERROR   -  Page 'blog/media.md' in 'tags_extra_files' must list tag identifiers as a list.
```

### Related links

- https://squidfunk.github.io/mkdocs-material/setup/setting-up-tags/#+tags.tags_extra_files

### Reproduction

To be provided if required

### Steps to reproduce

1. have a repo with tags_extra_files

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Oops, that's clearly a bug in the type declaration that came with one of the latest releases. Looking into it asap.

**Comment by squidfunk** [comment]
Released as part of [9.2.6+insiders-4.40.2](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.40.2).
